lib/infra/firebase_notifications/firebase_notifications_service.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final AutoDisposeProvider<FirebaseNotificationsService>
+    firebaseNotificationsService =
+    AutoDisposeProvider<FirebaseNotificationsService>(
+        (AutoDisposeProviderRef<FirebaseNotificationsService> ref) =>
+            FirebaseNotificationsService());
+
+class FirebaseNotificationsService {
+  /// Notification permission prompt
+  Future<bool> askForPermissions() async {
+    NotificationSettings notificationSettings =
```

Comment:
we should use a plugin like premission_handler because permission can be permanently denied to which we would open system settings